charts/plugin-site/templates/deployment.yaml

```diff
@@ -22,9 +22,79 @@ spec:
         {{- toYaml . | nindent 8 }}
     {{- end }}
       containers:
-        - name: {{ .Chart.Name }}
-          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
-          imagePullPolicy: {{ .Values.image.pullPolicy }}
+        - name: {{ .Chart.Name }}-frontend
+          image: "{{ .Values.frontend.image.repository }}:{{ .Values.frontend.image.tag }}"
+          imagePullPolicy: {{ .Values.frontend.image.pullPolicy }}
+          {{ if .Values.frontend.image.oldMode -}}
+          workingDir: /plugins
+          command: ["yarn"]
+          args: ["server"]
+          {{ end -}}
+          ports:
+              - name: frontend
+                containerPort: {{ .Values.frontend.port }}
+                protocol: TCP
+          env:
+            - name: REST_API_URL
+              value: {{ .Values.restApiUrl }}
+          {{ if not .Values.frontend.image.oldMode -}}
+          livenessProbe:
+              httpGet:
+                  path: /
+                  port: {{ .Values.frontend.port }}
+                  scheme: HTTP
+              initialDelaySeconds: 5
+              timeoutSeconds: 5
+          readinessProbe:
+              httpGet:
+                  path: /
+                  port: {{ .Values.frontend.port }}
+                  scheme: HTTP
+              initialDelaySeconds: 5
+              timeoutSeconds: 5
+          {{ end -}}
+          resources:
+            {{- toYaml .Values.frontend.resources | nindent 12 }}
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+---
+apiVersion: apps/v1
```

Comment:
I think it would be cleaner if each deployment was in its own file, more obvious where to look to find the resource to me